### PR TITLE
Device Support: TS110F Lonsonho

### PR DESCRIPTION
**Pull request by umairyounus**
Device Support: TS110F Lonsonho - Smart Dimmer Switch Module With Neutral

### CODE REVIEW

**Comment by Koenkk** [comment]
Could you provide a link to this device?

**Comment by umairyounus** [comment]
Here is the link:
https://www.aliexpress.com/item/4001279149071.html?spm=a2g0s.9042311.0.0.27424c4d1t8mTl

**Comment by Koenkk** [comment]
Thanks!

File: devices.js [hunk 1]
```diff
@@ -10228,6 +10228,25 @@ const devices = [
             await configureReporting.onOff(endpoint);
         },
     },
+    {
+        zigbeeModel: ['TS110F'],
```
Comment (ginkage) [inline devices.js:10232]: Hi!
Could you, please, follow-up with a change that replaces zigbeeModel with something like:
`        fingerprint: [{modelID: 'TS110F', manufacturerName: '_TYZB01_qezuin6k'}],`
(the fingerprint above is for 1-gang dimmer, which has the same model ID). You can find the manufacturerName in the logs when pairing:
```
debug 2020-08-25 18:36:09: Received Zigbee message from '0xbc33acfffe6d81d6', type 'readResponse', cluster 'genBasic', data '{"modelId":"TS110F"}' from endpoint 1 with groupID 0
debug 2020-08-25 18:36:09: Received Zigbee message from '0xbc33acfffe6d81d6', type 'readResponse', cluster 'genBasic', data '{"manufacturerName":"_TYZB01_qezuin6k"}' from endpoint 1 with groupID 0
```

**Comment by ginkage** [comment]
Hi!
Could you, please, follow-up with a change that replaces zigbeeModel with something like:
`fingerprint: [{modelID: 'TS110F', manufacturerName: '_TYZB01_qezuin6k'}],`
(the fingerprint above is for 1-gang dimmer, which has the same model ID). You can find the manufacturerName in the logs when pairing:
```
debug 2020-08-25 18:36:09: Received Zigbee message from '0xbc33acfffe6d81d6', type 'readResponse', cluster 'genBasic', data '{"modelId":"TS110F"}' from endpoint 1 with groupID 0
debug 2020-08-25 18:36:09: Received Zigbee message from '0xbc33acfffe6d81d6', type 'readResponse', cluster 'genBasic', data '{"manufacturerName":"_TYZB01_qezuin6k"}' from endpoint 1 with groupID 0
```
In the meantime I'll prepare a PR to support 1-gang dimmer.

**Comment by ginkage** [comment]
On that topic, convertor for the 1-gang variant seems to look like this:
```
    {
        fingerprint: [{modelID: 'TS110F', manufacturerName: '_TYZB01_qezuin6k'}],
        model: 'QS-Zigbee-D02-TRIAC-LN',
        vendor: 'Lonsonho',
        description: '1 gang smart dimmer switch module with neutral',
        extend: generic.light_onoff_brightness_colortemp,
    },
```

**Comment by Koenkk** [comment]
If I understand correctly your problem is that you have a 1 gang carrying the same model? (`TS110F`)?

**Comment by ginkage** [comment]
Yes.

**Comment by Koenkk** [comment]
@umairyounus can you share the data/database.db entry of your 2 gang switch?

**Comment by ginkage** [comment]
Filed #1505 for 1-gang support.

**Comment by umairyounus** [comment]
Here you go:

```javascript
{
  "id": 20,
  "type": "Router",
  "ieeeAddr": "0xbc33acfffe6563ae",
  "nwkAddr": 7365,
  "manufId": 4098,
  "manufName": "_TYZB01_v8gtiaed",
  "powerSource": "Mains (single phase)",
  "modelId": "TS110F",
  "epList": [
    1,
    2
  ],
  "endpoints": {
    "1": {
      "profId": 260,
      "epId": 1,
      "devId": 257,
      "inClusterList": [
        0,
        4,
        5,
        6,
        8
      ],
      "outClusterList": [
        25,
        10
      ],
      "clusters": {
        "genBasic": {
          "attributes": {
            "modelId": "TS110F",
            "manufacturerName": "_TYZB01_v8gtiaed",
            "powerSource": 1,
            "zclVersion": 3,
            "appVersion": 65,
            "stackVersion": 0,
            "hwVersion": 1,
            "dateCode": ""
          }
        },
        "genOnOff": {
          "attributes": {
            "onOff": 0,
            "onTime": 0
          }
        },
        "genLevelCtrl": {
          "attributes": {
            "64512": 1023,
            "currentLevel": 254
          }
        }
      },
      "binds": [
        {
          "cluster": 6,
          "type": "endpoint",
          "deviceIeeeAddress": "0x00124b001caaacb3",
          "endpointID": 1
        },
        {
          "cluster": 8,
          "type": "endpoint",
          "deviceIeeeAddress": "0x00124b001caaacb3",
          "endpointID": 1
        }
      ]
    },
    "2": {
      "profId": 260,
      "epId": 2,
      "devId": 257,
      "inClusterList": [
        4,
        5,
        6,
        8
      ],
      "outClusterList": [
        
      ],
      "clusters": {
        "genOnOff": {
          "attributes": {
            "onOff": 0,
            "onTime": 0
          }
        },
        "genLevelCtrl": {
          "attributes": {
            "64512": 1023,
            "currentLevel": 111
          }
        }
      },
      "binds": [
        
      ]
    }
  },
  "appVersion": 65,
  "stackVersion": 0,
  "hwVersion": 1,
  "dateCode": "",
  "zclVersion": 3,
  "interviewCompleted": true,
  "meta": {
    
  },
  "lastSeen": 1598027156327
}

**Comment by ginkage** [comment]
Perfect, thank you! I'll file a PR.

**Comment by ginkage** [comment]
I've updated #1505 with the fingerprint matching. @Koenkk, please take a look.

**Pull request merged**